### PR TITLE
Add check for HTS Genotype schema to split_multi_hts

### LINKED ISSUE #2755: split_multi_hts on hardcalled VDS
To report a bug, fill in the information below. 
For support and feature requests, please use the discussion forum: http://discuss.hail.is/

-------------------------------------------------------------------------------------------

### Hail version:
devel-64a1fac

### What you did:
run split_multi_hts on a VDS with only GT genotype field

### What went wrong (all error messages here, including the full java stack trace):
```
is.hail.utils.HailException: Struct has no field `AD'
  Available fields:
    GT: Call
<input>:4:  newad = if (isDefined(g.AD))
                                   ^
                    at is.hail.utils.ErrorHandling$class.fatal(ErrorHandling.scala:6)
                    at is.hail.utils.package$.fatal(package.scala:27)
                    at is.hail.expr.ParserUtils$.error(Parser.scala:33)
                    at is.hail.expr.AST.parseError(AST.scala:255)
                    at is.hail.expr.Select.typecheckThis(AST.scala:333)
                    at is.hail.expr.AST.typecheckThis(AST.scala:246)
                    at is.hail.expr.AST.typecheck(AST.scala:252)
                    at is.hail.expr.AST$$anonfun$typecheck$1.apply(AST.scala:251)
                    at is.hail.expr.AST$$anonfun$typecheck$1.apply(AST.scala:251)
                    at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
                    at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
                    at is.hail.expr.AST.typecheck(AST.scala:251)
                    at is.hail.expr.Apply.typecheck(AST.scala:663)
                    at is.hail.expr.AST$$anonfun$typecheck$1.apply(AST.scala:251)
                    at is.hail.expr.AST$$anonfun$typecheck$1.apply(AST.scala:251)
                    at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
                    at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
                    at is.hail.expr.AST.typecheck(AST.scala:251)
                    at is.hail.expr.Let$$anonfun$typecheck$18.apply(AST.scala:850)
                    at is.hail.expr.Let$$anonfun$typecheck$18.apply(AST.scala:849)
                    at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:733)
                    at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
                    at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
                    at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:732)
                    at is.hail.expr.Let.typecheck(AST.scala:849)
                    at is.hail.expr.Parser$$anonfun$13.apply(Parser.scala:204)
                    at is.hail.expr.Parser$$anonfun$13.apply(Parser.scala:203)
                    at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
                    at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
                    at scala.collection.immutable.List.foreach(List.scala:381)
                    at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
                    at scala.collection.immutable.List.map(List.scala:285)
                    at is.hail.expr.Parser$.parseNamedExprs(Parser.scala:203)
                    at is.hail.expr.Parser$.parseAnnotationExprs(Parser.scala:123)
                    at is.hail.methods.ExprAnnotator.<init>(SplitMulti.scala:14)
                    at is.hail.methods.SplitMulti.<init>(SplitMulti.scala:179)
                    at is.hail.methods.SplitMulti$.apply(SplitMulti.scala:149)
                    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
                    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                    at java.lang.reflect.Method.invoke(Method.java:498)
                    at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:237)
                    at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
                    at py4j.Gateway.invoke(Gateway.java:280)
```

### PR DESCRIPTION
**Pull request by jigold**
Fixes #2755

### CODE REVIEW

File: python/hail/api1/dataset.py [hunk 1]
```diff
@@ -4988,6 +4988,16 @@ def split_multi_hts(self, keep_star=False, left_aligned=False):
         :rtype: :py:class:`.VariantDataset`
         """
 
+        hts_genotype_schema = TStruct._from_java(Env.hail().variant.Genotype.htsGenotypeType().deepOptional())
```
Comment (tpoterba) [inline python/hail/api1/dataset.py]: shouldn't this fix be on methods/split_multi_hts? this stuff can fall to ruins

File: python/hail/api1/tests.py [hunk 2]
```diff
@@ -907,3 +907,10 @@ def test_pca(self):
         _, _, loadings = (hc.balding_nichols_model(3, 100, 100)
                           .pca('g.GT.nNonRefAlleles()', k=2))
         self.assertEqual(loadings, None)
+
+    def test_split_multi_hts(self):
+        vds = hc.import_vcf("src/test/resources/sample.vcf")
+        vds = vds.annotate_genotypes_expr("g = {GT: g.GT}")
+        self.assertRaisesRegexp(FatalError,
+                                "split_multi_hts: genotype_schema must be the HTS genotype schema",
+                                vds.split_multi_hts)
```
Comment (konradjk) [inline python/hail/api1/tests.py]: missing parens after split_multi_hts

Comment (jigold) [reply python/hail/api1/tests.py]: Thats the correct signature for `assertRaisesRegexp`

Comment (tpoterba) [reply python/hail/api1/tests.py]: ya, same as `lambda: vds.split_multi_hts()`

Comment (konradjk) [reply python/hail/api1/tests.py]: Oh my mistake - I thought I saw one above that was different, but I'm just losing my mind. Carry on

**Pull request merged**